guppylang/compiler/expr_compiler.py

```diff
@@ -753,6 +759,21 @@ def array_comprehension_init_func(ctx: CompilerContext) -> hf.Function:
     return func
 
 
+def array_unwrap_elem(ctx: CompilerContext) -> hf.Function:
+    """Returns the Hugr function that is used to unwrap the elements in an option array
+    to turn it into a regular array."""
+    v = ht.Variable(0, ht.TypeBound(ht.TypeBound.Any))
+    sig = ht.PolyFuncType(
+        params=[ht.TypeTypeParam(ht.TypeBound.Any)],
+        body=ht.FunctionType([ht.Option(v)], [v]),
+    )
+    func, already_defined = ctx.declare_global_func(ARRAY_UNWRAP_ELEM, sig)
+    if not already_defined:
```

Comment:
guess not relevant to this PR but I would have expected the declare function to return `function | None` for this kind of pattern and to condition off that.

I assume there is some valid usage for non-null function with `already_defined` being False?

Reply:
If `already_defined = False`, the returned `func` is a builder to construct the function. This way, we don't need to separately put the defined function back into the context once we're done building